pkg/sfu/streamallocator.go

```diff
@@ -686,16 +687,13 @@ func (s *StreamAllocator) runStateStable(event Event) {
 func (s *StreamAllocator) runStateDeficient(event Event) {
 	switch event.Signal {
 	case Signal_ADD_TRACK:
-		// wait for layers add/remove signal
+		s.allocate()
 	case Signal_REMOVE_TRACK:
 		s.allocate()
 	case Signal_ESTIMATE_INCREASE:
 		// as long as estimate is increasing, keep going.
-		// Switch to STABLE state if estimate exceeds optimal bandwidth needed.
-		if s.getChannelCapacity() > s.getOptimalBandwidthUsage() {
-			s.resetBoost()
-			s.setState(State_STABLE)
-		}
+		// try an allocation to check if state can move to STABLE
+		s.allocate()
```

Comment:
I can't remember exactly this condition, but it went to STABLE when it should not have under some condition. So, an allocation is a clean way to determine if state can go to STABLE.

Again, this is one of those places where we need to look deep if an allocation is necessary or if we can avoid it.